### DISCUSSION TITLE
how to know if the user have administrator permission in latest version

### DISCUSSION THREAD

**Discussion by plaraje** [discordjs/discord.js#8570] [Q&A] [answered]
Hi, djs changed a lot since i stopped making bots, now that im back a lot of thinks changed and i dont find the way to check if msg.member have "ADMINISTRATOR" permissions

**Comment by d4kw1n** [comment]
I think you can use `msg.member.permissions.has("ADMINISTRATOR");`
in the try catch to do this. I don't know why this command now shows the error `RangeError [BitFieldInvalid]: Invalid bitfield flag or number: ADMINISTRATOR`, but if try catch it still partially handles this.

**Reply by imranbarbhuiya** [reply]
You need to use https://discord.js.org/#/docs/discord.js/main/class/PermissionsBitField?scrollTo=s-Flags instead of string `ADMINISTRATOR`. The try  and catch will just ignore the admin check lol

**Reply by d4kw1n** [reply]
I only show how to check in case of true, and if there is an error, oh no sorry lol. Also thank you for sharing more

**Reply by plaraje** [reply]
ok thanks i'll try

**Comment by d4kw1n** [comment]
I did some research and found a way you can use and give me feedback.
`const raw = msg.guild.members.fetch(msg.author.id).then((rel) => { return rel });
            const checkAdmin = () => {
                raw.then((member) => {
                    if (member.permissions.has(PermissionsBitField.Flags.ManageGuild)) {
                        ...
                    }
                    else{
                        ...
                    }
                })
            };
            checkAdmin();
`

**Reply by almostSouji** [reply]
yeah, no:
` msg.guild.members.fetch(msg.author.id)` obsolete, prefer `message.member`
`.then((rel) => { return rel })` does absolutely nothing, as the promise returned is resolved and wrapped in a promise again.
I have no idea why you suddenly pull out an arrow function or use ManageGuild, which wasn't the question, to begin with.

**Comment by almostSouji** [comment, marked as answer]
For base permissions (guild level, default role and any roles the member might have):
```js
message.member.permissions.has(PermissionFlagsBits.Administrator);
```

For permissions after channel overwrites:
<https://discordjs.guide/popular-topics/permissions.html#permissions-after-overwrites>

If using command interactions, you can get the permissions of the member executing the command after overwrites through
<https://discord.js.org/#/docs/discord.js/14.3.0/class/CommandInteraction?scrollTo=memberPermissions> instead. Checking for the flag works exactly the same as above.

You can find more information on Permissions in general in our guide at <https://discordjs.guide/popular-topics/permissions.html>

**Reply by d4kw1n** [reply]
Oh thank you so much for helping me solve this problem. I just started with Discordjs, tried searching and got answers only for older versions, thanks to you this is easier.  🤗

**Reply by plaraje** [reply]
Yay i think is this one let me try, thanks
